server/sendq.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020-2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"strconv"
+	"sync"
+	"time"
+)
+
+type outMsg struct {
+	subj string
+	rply string
+	hdr  []byte
+	msg  []byte
+	next *outMsg
+}
+
+type sendq struct {
+	mu    sync.Mutex
+	mch   chan struct{}
+	head  *outMsg
+	tail  *outMsg
+	s     *Server
+	msgs  int
```

Comment:
Does not look like you are doing anything with msgs and bytes, so maybe remove?

Reply:
I track them but do not act on them atm.. Will remove.